Source/SwiftLintFramework/Rules/Lint/DuplicateConditionsRule.swift

```diff
@@ -0,0 +1,214 @@
+                  foo()
+                case \"a\" where y == "t":
+                  bar()
+                }
+            """),
+            Example("""
+                if let x = maybeAbc {
+                  foo()
+                } else if let x = maybePqr {
+                  bar()
+                }
+            """),
+            Example("""
+                if let x = maybeAbc, let z = x.maybeY {
+                  foo()
+                } else if let x = maybePqr, let z = x.maybeY {
+                  bar()
+                }
+            """),
+            Example("""
+                if case .p = x {
+                  foo()
+                } else if case .q = x {
+                  bar()
+                }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+                if ↓x < 5 {
+                  foo()
+                } else if y == "s" {
+                  bar()
+                } else if ↓x < 5 {
+                  baz()
+                }
+            """),
+            Example("""
+                if ↓x < 5, y == "s" {
+                  foo()
+                } else if x < 10 {
+                  bar()
+                } else if ↓y == "s", x < 5 {
+                  baz()
+                }
+            """),
+            Example("""
+                switch x {
+                case ↓\"a\", \"b\":
+                  foo()
+                case \"c\", ↓\"a\":
+                  bar()
+                }
+            """),
+            Example("""
+                switch x {
+                case ↓\"a\" where y == "s":
+                  foo()
+                case ↓\"a\" where y == "s":
+                  bar()
+                }
+            """),
+            Example("""
+                if ↓let xyz = maybeXyz {
+                  foo()
+                } else if ↓let xyz = maybeXyz {
+                  bar()
+                }
+            """),
+            Example("""
+                if let x = maybeAbc, let z = x.maybeY {
+                  foo()
+                } else if let x = maybeAbc, let z = x.maybeY {
+                  bar()
+                }
+            """),
+            Example("""
+                if ↓#available(macOS 10.15, *) {
+                  foo()
+                } else if ↓#available(macOS 10.15, *) {
+                  bar()
+                }
+            """),
+            Example("""
+                if ↓case .p = x {
+                  foo()
+                } else if ↓case .p = x {
+                  bar()
+                }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension DuplicateConditionsRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: IfStmtSyntax) {
+            if let prevToken = node.previousToken,
+               prevToken.tokenKind == .elseKeyword {
```

Comment:
We typically check that like `node.parent.is(IfExprSyntax.self)` which is a bit more direct.

Reply:
Much nicer! Thank you